manager/download.go

```diff
@@ -293,74 +350,83 @@ func (d *downloader) download() (n int64, err error) {
 //
 // If this is the first worker, this operation also resolves the total number
 // of bytes to be read so that the worker manager knows when it is finished.
-func (d *downloader) downloadPart(ch chan dlchunk) {
+func (d *downloader) downloadPart(ch chan dlchunk, toWrite chan *dlchunk) {
 	defer d.wg.Done()
+
+	allowedBuffered := d.cfg.PartSize * int64(d.cfg.MaxBufferedParts)
+
 	for {
 		chunk, ok := <-ch
 		if !ok {
 			break
 		}
+
+		for {
+			if d.getErr() != nil {
+				break
+			}
+			// if we are attempting to download a part that is too far ahead, then wait a bit
```

Comment:
We mentioned this in-person, but it would be nice to at least add a `// TODO:` here to use some sort of channel to "wake up" goroutines that are too far past the flushed index.

Reply:
yeah I'll add a TODO.